lib/matplotlib/artist.py

```diff
@@ -1417,6 +1419,67 @@ def set_mouseover(self, mouseover):
     mouseover = property(get_mouseover, set_mouseover)  # backcompat.
 
 
+class ColorizingArtist(Artist):
+    def __init__(self, colorizer):
+        """
+        Parameters
+        ----------
+        colorizer : `colorizer.Colorizer`
+        """
+        if not isinstance(colorizer, Colorizer):
+            raise ValueError("A `mpl.colorizer.Colorizer` object must be provided")
+
+        Artist.__init__(self)
+
+        self._A = None
+
+        self.colorizer = colorizer
+        self._id_colorizer = self.colorizer.callbacks.connect('changed', self.changed)
+        self.callbacks = cbook.CallbackRegistry(signals=["changed"])
+
+    def set_array(self, A):
+        """
+        Set the value array from array-like *A*.
+
+        Parameters
+        ----------
+        A : array-like or None
+            The values that are mapped to colors.
+
+            The base class `.ColorizingArtist` does not make any assumptions on
+            the dimensionality and shape of the value array *A*.
+        """
+        if A is None:
+            self._A = None
+            return
+
+        A = cbook.safe_masked_invalid(A, copy=True)
+        if not np.can_cast(A.dtype, float, "same_kind"):
+            raise TypeError(f"Image data of dtype {A.dtype} cannot be "
+                            "converted to float")
+
+        self._A = A
+        if not self.norm.scaled():
```

Comment:
are we sure that this will always have a `.norm`?

Reply:
I'm thinking it will always have a norm, as I think that simplifies the implementation. 
However, that norm might be `NoNorm`.

The implementation here mirrors that of `ScalarMappable`, which always has a `Normalize` and `Colormap` object, and I think this is sensible. For images that are already RGB(A) a norm (and cmap) is technically unnecessary, but I believe the overhead of creating them is negligible.